# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.cpp
# Commit: 7d5683180ac3
# Full Hash: 7d5683180ac3f9e7cc96d68dde27b8b48642a7ac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r a48aff5f2555 -r 7d5683180ac3 dom/script/ScriptSettings.cpp
--- a/dom/script/ScriptSettings.cpp	Thu Feb 17 15:57:52 2022 +0000
+++ b/dom/script/ScriptSettings.cpp	Thu Feb 17 15:57:52 2022 +0000
@@ -7,7 +7,6 @@
 #include "mozilla/dom/ScriptSettings.h"
 
 #include <utility>
-#include "LoadedScript.h"
 #include "MainThreadUtils.h"
 #include "js/CharacterEncoding.h"
 #include "js/CompilationAndEvaluation.h"
@@ -21,6 +20,8 @@
 #include "js/Warnings.h"
 #include "js/Wrapper.h"
 #include "js/friend/ErrorMessages.h"
+#include "js/loader/LoadedScript.h"
+#include "js/loader/ScriptLoadRequest.h"
 #include "jsapi.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/BasePrincipal.h"
@@ -33,7 +34,6 @@
 #include "mozilla/dom/BindingUtils.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/Element.h"
-#include "mozilla/dom/ScriptLoadRequest.h"
 #include "mozilla/dom/WorkerCommon.h"
 #include "nsContentUtils.h"
 #include "nsDebug.h"
@@ -55,7 +55,8 @@
   // NOTE: The result of this is only used by DevTools for matching sources, so
   // it is safe to silently ignore any errors and return nullptr for them.
 
-  LoadedScript* script = static_cast<LoadedScript*>(aPrivateValue.toPrivate());
+  JS::loader::LoadedScript* script =
+      static_cast<JS::loader::LoadedScript*>(aPrivateValue.toPrivate());
 
   JS::Rooted<JS::Value> elementValue(aCx);
   {